.github/workflows/linting-and-tests.yml

```diff
@@ -58,21 +58,19 @@ jobs:
       - name: Checkout project
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
-          persist-credentials: true
+          persist-credentials: false
 
       - name: Setup environment
         uses: ./.github/actions/setup-env
 
-      - name: Codegen
-        run: pnpm codegen:all
-
-      - name: Check for changes and commit if needed
+      - name: Check for codegen'd changes
         run: |
+          pnpm codegen:all
+
           # Check both unstaged and staged changes
           if ! git diff --exit-code || ! git diff --cached --exit-code; then
-            git add .
-            git commit -m "chore: update generated code"
-            git push --no-verify
```

Comment:
let me update this + the below portion of this to use our github app (currently the `git commit/push` here doesn't work fullly)